src/hypothesis/internal/charmap.py

```diff
@@ -106,6 +107,36 @@ def categories():
     return tuple(_categories)
 
 
+def validate_categories(cats, name=None):
+    """Return a tuple of Unicode categories in a normalised order.
+
+    This function expands one-letter designations of a major class to include
+    all subclasses:
+
+    >>> validate_categories(['N'])
+    ('Nd', 'Nl', 'No')
+
+    See section 4.5 of the Unicode standard for more on classes:
+    https://www.unicode.org/versions/Unicode10.0.0/ch04.pdf
+
+    If the collection ``cats`` includes any elements that do not represent a
+    major class or a class with subclass, InvalidArgument is raised.
+
+    """
+    if cats is None:
+        return None
+    major_classes = ('L', 'M', 'N', 'P', 'S', 'Z', 'C')
+    cs = categories()
+    out = set(cats)
+    for c in cats:
+        if c in major_classes:
+            out.discard(c)
+            out.update(x for x in cs if x.startswith(c))
+        elif c not in cs:
+            raise InvalidArgument('%r is not a valid Unicode category' % (c,))
```

Comment:
I'm kinda on the fence about whether this should be a warning instead of an error. Sadly I think it probably should. 😞